src/lib/components/Cytoscape.react.js

```diff
@@ -173,70 +173,80 @@ class Cytoscape extends Component {
+               the selectedNodeData prop.
+               */
             const nodeData = selectedNodes.map((el) => el.data());
 
-            if (typeof this.props.setProps === 'function') {
-                this.props.setProps({
-                    selectedNodeData: nodeData,
-                });
-            }
+            this.props.setProps({
+                selectedNodeData: nodeData,
+            });
         }, SELECT_THRESHOLD);
 
         const sendSelectedEdgesData = _.debounce(() => {
             const edgeData = selectedEdges.map((el) => el.data());
 
-            if (typeof this.props.setProps === 'function') {
-                this.props.setProps({
-                    selectedEdgeData: edgeData,
-                });
-            }
+            this.props.setProps({
+                selectedEdgeData: edgeData,
+            });
         }, SELECT_THRESHOLD);
 
         // /////////////////////////////////////// EVENTS //////////////////////////////////////////
         cy.on('tap', 'node', (event) => {
             const nodeObject = this.generateNode(event);
 
-            if (typeof this.props.setProps === 'function') {
-                this.props.setProps({
-                    tapNode: nodeObject,
-                    tapNodeData: nodeObject.data,
-                });
-            }
+            this.props.setProps({
+                tapNode: nodeObject,
+                tapNodeData: Object.assign({}, nodeObject.data, {
+                    timeStamp: nodeObject.timeStamp,
+                }),
+            });
         });
 
         cy.on('tap', 'edge', (event) => {
             const edgeObject = this.generateEdge(event);
 
-            if (typeof this.props.setProps === 'function') {
-                this.props.setProps({
-                    tapEdge: edgeObject,
-                    tapEdgeData: edgeObject.data,
-                });
-            }
+            this.props.setProps({
+                tapEdge: edgeObject,
+                tapEdgeData: Object.assign({}, edgeObject.data, {
+                    timeStamp: edgeObject.timeStamp,
+                }),
+            });
         });
 
         cy.on('mouseover', 'node', (event) => {
-            if (typeof this.props.setProps === 'function') {
-                this.props.setProps({
-                    mouseoverNodeData: event.target.data(),
-                });
-            }
+            this.props.setProps({
+                mouseoverNodeData: Object.assign({}, event.target.data(), {
+                    timeStamp: event.timeStamp,
+                }),
+            });
         });
 
         cy.on('mouseover', 'edge', (event) => {
-            if (typeof this.props.setProps === 'function') {
+            this.props.setProps({
+                mouseoverEdgeData: Object.assign({}, event.target.data(), {
+                    timeStamp: event.timeStamp,
+                }),
+            });
+        });
+
+        if (this.props.clearOnUnhover === true) {
```

Comment:
I'd recommend putting this `if` statement *inside* each event function — that way, if the value of `clearOnUnhover` gets changed by a callback, the behavior will change accordingly:

```javascript
        
            cy.on('mouseout', 'node', (_) => {
                if (this.props.clearOnUnhover === true) {
                     this.props.setProps({
                         mouseoverNodeData: null,
                     });
                }
             });
```